projects/lights_control/inc/lights_blinker.h

```diff
@@ -0,0 +1,40 @@
+#pragma once
+#include "event_queue.h"
+#include "soft_timer.h"
+// Blinking requires soft_timers and event_queue to be initialized.
+
+// This module aids with the blinking functionality that some lights require such as signal lights.
+// An instance of this module is used to periodically raise events. The data field of the event
+// will switch between 1 and 0, corresponding to ON and OFF states. Blinking begins with ON state.
+
+// State definitions for blinker.
+typedef enum {
+  LIGHTS_BLINKER_STATE_OFF = 0,
+  LIGHTS_BLINKER_STATE_ON,
+  NUM_LIGHTS_BLINKER_STATES
+} LightsBlinkerState;
+
+typedef uint32_t LightsBlinkerDuration;
+
+// Blinker members.
+typedef struct LightsBlinker {
+  EventID event_id;
+  SoftTimerID timer_id;
+  LightsBlinkerState state;
+  LightsBlinkerDuration duration_ms;  // Duration, in milliseconds
+} LightsBlinker;
+
+// Initializes the blinker.
+StatusCode lights_blinker_init(LightsBlinker *blinker);
+
+// Starts generating events.
+StatusCode lights_blinker_on(LightsBlinker *blinker, LightsBlinkerDuration duration_ms, EventID id);
```

Comment:
Making a new comment to make sure you see this: Why are the duration and event ID passed here? Doesn't it make more sense to set it up in your init?

Reply:
@ckitagawa suggested that I make the api as flexible as possible, I initially had it in my init.

Reply:
And as for `event ID`, I think it'll be better to pass it here, since I want to use only one blinker to generate multiple event types. In `lights_signal` for example, I'm using one blinker to generate left signal blink, right signal blink, and hazard blink events. 

It would be possible to use one blinker instance still if I had it in my `lights_blinker_init` function, but I think in that case I'd have to call `lights_blinker_init` multiple times on my one `LightsBlinker` instance, which I think would be weird, cuz init functions are only called once, to initialize the blinker.

Reply:
What's a potential use case that would benefit from this?

Reply:
I see your point. It is in fact only used for syncing purposes too. 
How about `lights_blinker_sync_reset' ?

Reply:
I don't think reset is the right term. Reset means restoring to initial state or clearing, which this isn't doing.

Reply:
Ah yeah. This is more like, updating the timer. How about `lights_blinker_sync_update`?

Reply:
`sync_update` doesn't really tell you what you're updating. I think a name that's more descriptive of the behavior would be good. Maybe `sync_on` or something similar?

Reply:
I think that's good. `sync_on` is both short, and indicates that both boards are getting synced to an `ON` state.